public-ui/src/app/highlight-search/highlight-search.component.ts

```diff
@@ -21,20 +21,26 @@ export class HighlightSearchComponent implements OnChanges {
       searchWords = searchWords.map(word => word.replace(/[&!^\/\\#,+()$~%.'":*?<>{}]/g, ''));
       this.matchString = new RegExp(searchWords.join('|'));
     }
-    this.words = this.text.split(' ');
-  }
-  public highlight(word) {
-    const matches = word.match(new RegExp(this.matchString, 'g'));
-    const splits = word.split(new RegExp(this.matchString));
-    const temp = [];
-    if (matches) {
+    const matches = this.text.match(new RegExp(this.matchString, 'gi'));
+    const splits = this.text.split(new RegExp(this.matchString, 'i'));
+    if (matches && this.searchTerm) {
       for (let i = 0; i < matches.length; i++) {
-        temp.push(splits[i], matches[i]);
+        // did not want to highlight whole of disorder of (, when searched by disorder
```

Comment:
I'm confused - why would a matched value contain spaces unless the search term also contained spaces? I don't see why it's necessary to split on space at this point.